core/org.openjdk.jmc.flightrecorder.rules.jdk/src/main/java/org/openjdk/jmc/flightrecorder/rules/jdk/memory/GcPauseTargetRule.java

```diff
@@ -0,0 +1,166 @@
+import org.openjdk.jmc.flightrecorder.rules.util.RulesToolkit.EventAvailability;
+import org.openjdk.jmc.flightrecorder.rules.util.RulesToolkit.RequiredEventsBuilder;
+
+/**
+ * Rule that looks at the G1 GC time and compares it to the max pause time.
+ */
+public class GcPauseTargetRule implements IRule {
+
+	private static final String GC_PAUSE_TARGET_RESULT_ID = "GcPauseTargetRule"; //$NON-NLS-1$
+
+	private static final Map<String, EventAvailability> REQUIRED_EVENTS = RequiredEventsBuilder.create()
+			.addEventType(JdkTypeIDs.GC_G1MMU, EventAvailability.AVAILABLE).build();
+
+	public static final TypedResult<IQuantity> EXCEEDED_PERCENT = new TypedResult<>("gcExceededPercent", //$NON-NLS-1$
+			"Pause target exceeded percent",
+			"The percentage of G1MMU events where the gcTime exceeded the pauseTarget.", UnitLookup.PERCENTAGE,
+			IQuantity.class);
+
+	public static final TypedResult<IQuantity> G1MMU_TOTAL = new TypedResult<>("g1mmuTotal", "Total G1MMU Events", //$NON-NLS-1$
+			"The number of G1MMU events that occured during the recording.", UnitLookup.NUMBER, IQuantity.class);
+
+	public static final TypedResult<IQuantity> G1MMU_EXCEEDED = new TypedResult<>("g1mmuExceeded", //$NON-NLS-1$
+			"Exceeded pause targets", "The number of G1MMU events where the gcTime exceeded the pauseTarget.",
+			UnitLookup.NUMBER, IQuantity.class);
+
+	public static final TypedResult<IQuantity> PAUSE_TARGET = new TypedResult<>("pauseTarget", "Pause target", //$NON-NLS-1$
+			"Max time allowed to be spent on GC during last time slice.", UnitLookup.NUMBER, IQuantity.class);
+
+	public final Collection<TypedResult<?>> RESULT_ATTRIBUTES = Arrays.<TypedResult<?>> asList(EXCEEDED_PERCENT,
+			G1MMU_TOTAL, G1MMU_EXCEEDED, PAUSE_TARGET);
+
+	@Override
+	public RunnableFuture<IResult> createEvaluation(
+		final IItemCollection items, final IPreferenceValueProvider preferenceValueProvider,
+		final IResultValueProvider dependencyResults) {
+		FutureTask<IResult> evaluationTask = new FutureTask<>(new Callable<IResult>() {
+			@Override
+			public IResult call() throws Exception {
+				IQuantity pauseTarget = items
+						.getAggregate((IAggregator<IQuantity, ?>) Aggregators.max(JdkAttributes.GC_PAUSE_TARGET));
+				IQuantity maxGcTime = items
+						.getAggregate((IAggregator<IQuantity, ?>) Aggregators.max(JdkAttributes.GC_TIME));
+				if (maxGcTime.compareTo(pauseTarget) < 1) {
+					return ResultBuilder.createFor(GcPauseTargetRule.this, preferenceValueProvider)
+							.setSeverity(Severity.OK).setSummary(Messages.getString(Messages.GcPauseTargetRule_TEXT_OK))
+							.addResult(PAUSE_TARGET, pauseTarget).build();
+				} else {
+					int exceededCount = 0;
+					IItemCollection g1mmuItems = items.apply(JdkFilters.GC_G1MMU);
+					for (IItemIterable item : g1mmuItems) {
+						for (IItem event : item) {
+							if (RulesToolkit.getValue(event, JdkAttributes.GC_TIME).compareTo(pauseTarget) < 1) {
+								exceededCount++;
+							}
+						}
+					}
```

Comment:
I'd prefer that we use a filter on the g1mmuItems that applies a comparison filter. E.g. ItemFilters.buildComparisonFilter(Kind.MORE, JdkAttributes.GC_TIME, pauseTarget) and then count the events in the resulting IItemCollection using an aggregator.